### DISCUSSION TITLE
Image saving duration for animations

### DISCUSSION THREAD

**Discussion by ghost** [vladmandic/sdnext#1240] [General]
I noticed something after experimenting with very low UniPC steps + ControlNet today. I very often use SD for animation. At some point I noticed that when the steps are very low and the iterations take a very short amount of time, there is a specific time limit that cannot be broken, and that is the time to start and end an image. Is it a CPU thing? I have a bit older Ryzen, but its a Six-Core one with fast RAM.

I have an M2 SSD, so storage is not the problem. My thinking was that for animation, it would be cool to have some kind of "fast mode" which reduces (possible) init / post-processing overhead - whatever that might be. If I can get a frame from 2 measly steps, but the time to init and end takes as long as 2 steps themselves each, at some point there is nothing more to gain, even with a 4090 (when images are saved and processed sequentially like loopback).

Maybe the fast mode could involve initializing a temporary RAM disk, if that makes it faster. I am just throwing around ideas - I was thinking that with a 4090 and creating animations the performance boost will not be very noticable because of this problem. For every other task it is perfect though. I did not dive too deep into the sampling process but I think increasing it/s for animations will not pay off in the future to a degree that I wanted it to (and everyone who makes animations in SD)

**Comment by vladmandic** [comment]
A, there is postprocessing that just can't be avoided - getting final denoiser image. Otherwise you're getting unusable tensors.

Second, UniPC works differently than any other sampler - it has special init and final steps which take longer. But on the other hand, all steps in between are faster and it takes less steps to complete image.

If you want purely linear steps, don't use UniPC.

**Reply by ghost** [reply]
Thanks for the reply. I feared it is something that cannot be changed. ControlNet especially adds a lot of overhead. Future models will probably have some kind of inter-frame consistency mode, or there will be another satisfying workaround that gets rid of the sequential nature of the loopback (atleast partially). Generating 3 or 4 frames at a time is much faster, but for that, every frame must know what the others are doing in advance...

In seedtravel it could be possible to adapt the process to process multiple images at once because the sequence is fixed, but in loopback, its a bit more difficult
